sdk/monitor/monitor-query/src/metricsClient.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { TokenCredential } from "@azure/core-auth";
+import {
+  MetricsListResponse,
+  MonitorManagementClient as GeneratedMetricsClient
+} from "./generated/metrics/src";
+import {
+  MetricDefinitionsListResponse,
+  MonitorManagementClient as GeneratedMetricsDefinitionsClient
+} from "./generated/metricsdefinitions/src";
+import {
+  MetricNamespacesListResponse,
+  MonitorManagementClient as GeneratedMetricsNamespacesClient
+} from "./generated/metricsnamespaces/src";
+import {
+  GetMetricDefinitionsOptions,
+  GetMetricNamespaces,
+  QueryMetricsOptions
+} from "./models/metricsModels";
+
+export class MetricsClient {
+  private _metricsClient: GeneratedMetricsClient;
+  private _definitionsClient: GeneratedMetricsDefinitionsClient;
+  private _namespacesClient: GeneratedMetricsNamespacesClient;
```

Comment:
why are these three generated clients anyway? Seems silly for a single operation apiece

Reply:
I'm inexperienced in this - they're separate swaggers today. I'm not sure if there's a way to say "merge all of these swaggers into a single generated client" if that's what you're suggesting.

Reply:
Seems like maybe a good autorest feature request? /cc @sarangan12 @joheredi

Reply:
To add to this - there are some duplicate contracts between them. So there's some possible savings on code generation there (perhaps not since it's still multiple endpoints).